game-runner/src/main/java/za/co/entelect/challenge/player/BotPlayer.java

```diff
@@ -125,12 +104,61 @@ private String runBot(String state, String textState) throws IOException {
             botConsoleOutput = botRunner.run();
         } catch (IOException e) {
             log.info("Bot execution failed: " + e.getLocalizedMessage());
+        } catch (TimeoutException e) { // IOException caught first, nothing happens after
+            log.info("Bot execution failed: " + e.getLocalizedMessage());
+            incrementTimeoutCounts();
         }
         return botConsoleOutput;
     }
 
+    static void writeRoundStateData(String playerSpecificJsonState,
+                                    String playerSpecificTextState,
+                                    String playerSpecificConsoleState,
+                                    String command,
+                                    int round,
+                                    String botConsoleOutput,
+                                    String playerName,
+                                    String saveStateLocation) throws IOException {
+        String mainDirectory = String.format("%s/%s", saveStateLocation, FileUtils.getRoundDirectory(round));
+        File fMain = new File(mainDirectory);
+        if (!fMain.exists()) {
+            fMain.mkdirs();
+        }
+
+        File f = new File(String.format("%s/%s", mainDirectory, playerName));
+        if (!f.exists()) {
+            f.mkdirs();
+        }
+
+        File fConsole = new File(String.format("%s/%s/%s", mainDirectory, playerName, "Console"));
+        if (!fConsole.exists()) {
+            fConsole.mkdirs();
+        }
+
+        FileUtils.writeToFile(String.format("%s/%s/%s", mainDirectory, playerName, "JsonMap.json"), playerSpecificJsonState);
+        FileUtils.writeToFile(String.format("%s/%s/%s", mainDirectory, playerName, "TextMap.txt"), playerSpecificTextState);
+        FileUtils.writeToFile(String.format("%s/%s/%s", mainDirectory, playerName, "PlayerCommand.txt"), command);
+        FileUtils.writeToFile(String.format("%s/%s/%s/%s", mainDirectory, playerName, "Console", "Console.txt"), playerSpecificConsoleState);
+        FileUtils.writeToFile(String.format("%s/%s/%s/%s", mainDirectory, playerName, "Console", "BotOutput.txt"), botConsoleOutput);
+    }
+
     @Override
     public void gameEnded(GameMap gameMap) {
+        String playerSpecificJsonState = jsonRenderer.render(gameMap, getGamePlayer());
+        String playerSpecificTextState = textRenderer.render(gameMap, getGamePlayer());
+        String playerSpecificConsoleState = consoleRenderer.render(gameMap, getGamePlayer());
 
+        try {
+            writeRoundStateData(playerSpecificJsonState,
+                    playerSpecificTextState,
+                    playerSpecificConsoleState,
+                    "",
+                    gameMap.getCurrentRound(),
+                    "",
+                    getName(),
+                    saveStateLocation);
+        } catch (IOException e) {
+            e.printStackTrace();
```

Comment:
Should be logging these exceptions so we can keep track of errors thrown during a match